### ISSUE TITLE
[1.19.2] Search Tree is rebuilt in main menu on resourcepack reload if it was previously built

### ISSUE THREAD

**Issue by DBotThePony** [MinecraftForge/MinecraftForge#9055]
**This is a vanilla bug/design issue, which affect mods on Forge, and not vanilla itself (aside from wasted CPU cycles).**

**tl;dr: Ultimate solution is to cut search tree rebuild if we are in main menu**

**Minecraft Version:** 1.19.2

**Forge Version:** 43.1.1

**Steps to Reproduce:**
 1. Start the game with mods which have items whose tooltips rely on server config values
 2. Load a world/Connect to compatible server
 3. Disconnect from world/server
 4. Force resourcepacks reload (by adding/removing/reordering resourcepack list)
 5. Observer game crash inside `appendHoverText` due to `Cannot get config value before config is loaded.`: [gist](https://gist.github.com/DBotThePony/08a1aa57b013565e17aff3b1a5911d60)

<details>

<summary>User's reaction:</summary>

https://user-images.githubusercontent.com/6339511/192080565-4474a006-7b6c-46b3-a010-6df37cd98087.mp4

</details>

**Description of issue:**

https://github.com/MinecraftForge/MinecraftForge/issues/7716 introduced exception throw when trying to access config value before config is loaded, which is ofc expected behavior.

However, there is an edge case with Search Tree and utilizing config values inside `appendHoverText`.

On resourcepacks reload, Creative menu Search Tree (package `net.minecraft.client.searchtree`) gets notified about resources being reloaded (`SearchRegistry#onResourceManagerReload`), however, search tree is, supposedly, empty, right after game start, so it does nothing in such case.

If we try to reload resourcepacks in main menu right after we started the game, nothing bad happens.

But, when we load into world/connect to a server, Search Tree is built, polling all `appendHoverText` of items. This does not cause any issues, since this happens after Server Config was loaded (at least in my dev client-only and client-server (connecting to dev env dedicated server from dev env client) cases).

But here is where things get weird; After we disconnect, Search Tree stays built, which means, once we force resourcepack reload, Search Tree will have to refresh it's values, even when we are not connected to world/server; In the latter case, Server configs are `null`, so anything trying to access server-only config values will produce an exception. But, since Search Tree is polling `appendHoverText` again, we gonna request server-only config value without server config being present.

Why cut Search Tree rebuild in *main menu*? Isn't this gonna break stuff? **No, it won't, because search tree is rebuilt any time you are connecting to world/server, so rebuilding Search Tree in main menu has absolutely no meaningful result (result is never used)**
